Sources/Common/Store/QueueInventoryMemoryStore.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+// In-memory store of the BQ inventory. This acts as a cache to unnecessary reading the file system from the file system.
+internal protocol QueueInventoryMemoryStore {
+    var inventory: [QueueTaskMetadata]? { get set }
+}
+
+// The in-memory store should be a singleton to be re-used during the lifecycle of the SDK.
+// Try to keep this class small because it's a singleton.
+//
+// sourcery: InjectRegister = "QueueInventoryMemoryStore"
+// sourcery: InjectSingleton
+class QueueInventoryMemoryStoreImpl: QueueInventoryMemoryStore {
+    @Atomic var inventory: [QueueTaskMetadata]?
```

Comment:
In early versions of the BQ, we had a cache of the inventory in-memory. The BQ was designed so that the inventory would be light-weight enough that it could be stored in memory, even if it's big. That's why the BQ uses separate files for inventory and task data, to keep the inventory light. 

From my own testing, I have not noticed the inventory being too large in memory to cause concern.